### PR TITLE
123 mac m1 installation

### LINKED ISSUE #123: Installing SOPHT for M1 Macs
Follow the installation instruction in README, then for Apple M1 chips do the following

```
pip uninstall pyfftw
conda install -c conda-forge pyfftw
conda install numba
```

Now Sopht should work!

### PR DESCRIPTION
**Pull request by sy-cui**
Fix #123 

The source of dependencies `PyFFTW` and `pystencils` are set to two temporary git repositories. 
`PyFFTW`: https://github.com/tp5uiuc/pyFFTW/tree/macos_fix
`pystencils`: https://github.com/sy-cui/pystencils/tree/macos_fix

The modifed `PyFFTW` looks up `fftw` library in `/opt/homebrew/lib` on macos platforms. 

In `pystencils`, the shared object (.so) file is not dynamically linked to `OpenMP` properly on Apple M1.  Restricting the `-fopenmp` compiler option to the linking stage alone seems to resolve this issue. For our purpose, it is also plausible to compile in one step instead of two
```bash
clang++ -shared -o object.so object.cpp -fopenmp -undefined dynamic_lookup [other options]
```
but we don't do it for now. 

A new `environment.yml` file is provided to integrate poetry into conda environment.
(curtesy of https://stackoverflow.com/questions/70851048/does-it-make-sense-to-use-conda-poetry)

Poetry version is restricted to below 1.6 to avoid a [known issue with version control](https://github.com/python-poetry/poetry/issues/8409).

Edit:
The changes regarding `-fopenmp` on pystencils is reverted as multithreading isn't enabled if the flag isn't passed. Instead the clang compiler flag `-Xclang` is removed as it prevents `-fopenmp` from being recognized as a regular command line option.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -20,15 +20,15 @@ packages = [
...
-pystencils = "^1.0"
+numba = ">=0.55"
 scipy = "^1.7.3"
 sympy = "^1.9"
-pyfftw = "^0.13.0"
 psutil = "^5.9.0"
 h5py = "^3.6"
 pyelastica = "^0.3"
-matplotlib = "^3.3.2"
+matplotlib = "3.7.1"
```
Comment (armantekinalp) [inline pyproject.toml:29]: Any particular reason why do we fix matplotlib version?

File: pyproject.toml [hunk 2]
```diff
@@ -20,15 +20,15 @@ packages = [
...
+pyfftw = { git="https://github.com/tp5uiuc/pyFFTW.git", branch="macos_fix" }
+pystencils = { git="https://github.com/sy-cui/pystencils.git", branch="macos_fix" }
```
Comment (armantekinalp) [inline pyproject.toml:31]: Is this repo private or dented I can't access?

Comment (sy-cui) [reply pyproject.toml:31]: Made the repo public. Should work now

File: pyproject.toml (hunk 1 above)
Comment (bhosale2) [reply pyproject.toml:29]: @sy-cui ?

**Review by bhosale2 — APPROVED** [review]
LGTM

File: pyproject.toml (hunk 1 above)
Comment (sy-cui) [reply pyproject.toml:29]: Matplotlib 3.8.0 seems to have removed some functionalities used in sopht. Also on my machine installing any other version with the in-environment poetry would also install `setuptools-scm`, which is unwanted and keeps complaining about not having a section in `pyproject.toml`.

**Comment by bhosale2** [comment]
@armantekinalp please take a look again and merge accordingly.

**Comment by armantekinalp** [comment]
I think this is not working yet. Songyuan was planning to test on different machines.  @sy-cui can you confirm?

**Comment by sy-cui** [comment]
After the most recent push everything should be fine now. I have tested on Bridges and Expanse and they install without issue.

**Comment by bhosale2** [comment]
@sy-cui  and @armantekinalp  current installation instructions on this branch didnt work on my machine:

Screenshot 2023-10-11 at 3 03 14 PM

**Comment by bhosale2** [comment]
ok I just realized the issue. The install instructions don't tell to install `fftw` via `brew` first. @sy-cui lets add it somewhere then? or what if we integrate it in `conda` env?

**Comment by sy-cui** [comment]
@armantekinalp @bhosale2 can we merge this?

**Review by bhosale2 — APPROVED** [review]
LGTM

**Pull request merged**